### PR TITLE
Use runtime directory instead of compile directory when running tests

### PR DESCRIPTION
**Pull request by pranavkm**
This change is [Reviewable](https://reviewable.io/reviews/dotnet/cli/2579)

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @pakrym \ @livarcocc. Fix for the issue here - https://github.com/dotnet/coreclr.xunit/pull/6#discussion_r60106482. For net45 applications, this points to the `bin\debug\net451\win7-x64` dir (for instance) and for netcore, this points to `bin\debug\netcoreapp1.0`. Pointing to the runtime directory is required for the xunit runner to find the assembly closure.

File: src/dotnet/commands/dotnet-test/Program.cs [hunk 1]
```diff
@@ -112,7 +113,11 @@ private static IEnumerable<ProjectContext> CreateProjectContexts(string projectP
                 throw new InvalidOperationException($"{projectPath} does not exist.");
             }
 
-            return ProjectContext.CreateContextForEachFramework(projectPath);
+            var runtimeIdentifiers = !string.IsNullOrEmpty(runtime) ?
+                new[] { runtime } :
+                PlatformServices.Default.Runtime.GetAllCandidateRuntimeIdentifiers();
+
+            return ProjectContext.CreateContextForEachFramework(projectPath, runtimeIdentifiers: runtimeIdentifiers);
```
Comment (pakrym) [inline src/dotnet/commands/dotnet-test/Program.cs]: Use `ProjectContext.CreateContextForEachFramework(projectPath).Select(c=>c.CreateRuntime(runtimeIdentifiers))` here

**Comment by pranavkm** [comment]
@pakrym  - updated

**Comment by pakrym** [comment]
:shipit:

**Pull request merged**